routes/weatherProviders/DarkSky.ts

```diff
@@ -18,11 +13,18 @@ async function getDarkSkyWateringData( coordinates: GeoCoordinates ): Promise< W
 		return undefined;
 	}
 
+	const periods = Math.min( 24, historicData.hourly.data.length );
+	const totals = { temp: 0, humidity: 0, precip: 0 };
+	for ( let index = 0; index < periods; index++ ) {
+		totals.temp += historicData.hourly.data[ index ].temperature;
+		totals.humidity += historicData.hourly.data[ index ].humidity;
+		totals.precip += historicData.hourly.data[ index ].precipIntensity
+	}
+
 	return {
-		// Calculate average temperature for the day using hourly data.
-		temp : historicData.hourly.data.reduce( ( sum, hourlyData ) => sum + hourlyData.temperature, 0 ) / historicData.hourly.data.length,
-		humidity: historicData.daily.data[ 0 ].humidity * 100,
-		precip: historicData.daily.data[ 0 ].precipIntensity * 24,
+		temp : totals.temp / periods,
```

Comment:
Suggest we limit precision of values to avoid growing the OS buffer. In App, we limit `temp` to 2 decimal point, `humidity` to nearest integer, `precip` to three decimal points.

Reply:
Values are rounded before being sent to the OS controller https://github.com/OpenSprinkler/OpenSprinkler-Weather/blob/master/routes/weather.ts#L316

Reply:
Yep, you are right. I got confused because WeatherData is passed straight through to App whereas WateringData is conformed before returning to OS. I wonder if precision/formatting/standardizing should be done in the WeatherProvider code e.g. icons ?